### DISCUSSION TITLE
How to control the number of vertices output by YOLOV8-SEG.PT

### DISCUSSION THREAD

**Discussion by jon-leun** [ultralytics/ultralytics#15629] [Q&A]
I know yolov8-segment will output many points to be a mask , but i only want fout points , how can i get that ? thanks!
Just like this. The left side is the original appearance, and the right side is the one I want
![mul](https://github.com/user-attachments/assets/22de11ea-47cc-4a59-84a1-626e7a83649a)

**Comment by jon-leun** [comment]
# YOLOv8.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 3, C2f, [512]] # 12

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 3, C2f, [256]] # 15 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]] # cat head P4
  - [-1, 3, C2f, [512]] # 18 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]] # cat head P5
  - [-1, 3, C2f, [1024]] # 21 (P5/32-large)

  - [[15, 18, 21], 1, Segment, [nc, 32, 256]] # Segment(P3, P4, P5)
  
  I try to change yolov8.0n head to 
    - [[15, 18, 21], 1, Segment, [nc, 32, 8]] # Segment(P3, P4, P5)
    It still not working

**Reply by glenn-jocher** [reply]
@jon-leun to achieve a four-point output, you can post-process the segmentation mask to fit a bounding box or use a contour detection algorithm to approximate the mask with fewer vertices. For more details, you might want to explore OpenCV's contour approximation methods.
